pulpcore/cli/common/generic.py

```diff
@@ -448,6 +448,65 @@ def pulp_option(*args: Any, **kwargs: Any) -> Callable[[FC], FC]:
     return click.option(*args, **kwargs)
 
 
+def resource_lookup_option(*args: Any, **kwargs: Any) -> Callable[[FC], FC]:
+    lookup_key: str = kwargs.pop("lookup_key", "name")
+    href_pattern: Optional[str] = kwargs.pop("href_pattern", None)
```

Comment:
I would pass the Context class instead of the href_pattern to this function ...

Reply:
I decided to pass there a list of context classes. Now, I am iterating over the list to find the best matching entity context. It was required to handle commands like:
```
pulp container repository show --repository foo
pulp container repository -t push show --repository foo
pulp container repository -t push show --repository pushed-foo
```

Reply:
I'm not getting that. All the contexts there should be subclasses of `PulpRepositoryContext` and therefore be the one we find. So `ctx.get_object(PulpRepositoryContext).HREF_PATTERN` is also the only pattern we need to match against, no?

Reply:
You were totally right. I updated the code a bit.